cmd/main.go

```diff
@@ -96,13 +88,12 @@ func main() {
 	}
 	defer nc.Close()
 
-	client, err := mqttConnect(svcName, *cfg, logger)
-	if err != nil {
-		log.Fatalf(err.Error())
-	}
+	redisClient := connectToRedis(cfg.Server.CacheURL, cfg.Server.CachePass, cfg.Server.CacheDB, logger)
+	msgCache := messages.NewRedisCache(redisClient)
 
-	svc := export.New(client, *cfg, logger)
-	svc.Start(svcName, nc)
+	svc := export.New(cfg, msgCache, logger)
+	svc.Start(svcName)
```

Comment:
If it's not in an infinite loop I would return and log thee error here. Eventually you can create const for returned error that you will use inside `Start`

Reply:
what error?

Reply:
The error that this function should return instead of log it in the function itself